src/trace/dma-trace.c

```diff
@@ -200,7 +200,7 @@ int dma_trace_init_complete(struct dma_trace_data *d)
 			      SOF_TASK_PRI_MED, trace_work, d, 0, 0);
 
 out:
-
+	trace_on();
 	return ret;
```

Comment:
So, calling dma_trace_init_complete will turn the tracing on BUT calling trace_init will not turn the tracing ON.

I feel that the normal sequence of enabling tracing should be:

```
dma_trace_init_complete
trace_on()
```

I think is called DMA trace.

And for the other trace we shold do:

```
trace_init()
trace_on()

```

Reply:
good point, moving and unifying the trace_on() invoking to the end of primary_core_init() looks better to me.